galaxy_ng/app/api/ui/viewsets/collection.py

```diff
@@ -24,35 +25,41 @@
 from galaxy_ng.app.common import pulp
 
 
-class CollectionFilter(CollectionVersionFilter):
+class CollectionFilter(pulp_ansible_viewsets.CollectionVersionFilter):
+    """pulp_ansible CollectionVersion filter for Collection viewset."""
     versioning_class = versioning.UIVersioning
     keywords = filters.CharFilter(field_name="keywords", method="filter_by_q")
 
 
 class CollectionViewSet(api_base.LocalSettingsMixin, pulp_ansible_galaxy_views.CollectionViewSet):
-    """ Viewset that uses CollectionVersion to display data for Collection."""
+    """Viewset that uses CollectionVersion's within distribution to display data for Collection's.
+
+    Collection list is filterable by CollectionFilter and includes latest CollectionVersion.
+
+    Collection detail includes CollectionVersion that is latest or via query param 'version'.
+    """
     versioning_class = versioning.UIVersioning
     filterset_class = CollectionFilter
     permission_classes = [access_policy.CollectionAccessPolicy]
 
-    # TODO(awcrosby): remove once pulp_ansible is_highest param filters by repo
-    # https://pulp.plan.io/issues/7428
-    def get_queryset(self):
-        """
-        Overrides to not use is_highest param and
-        ensures one collection version per collection is returned.
-        """
-        distro_content = self.get_distro_content(self.kwargs["path"])
+    def get_object(self):
+        """Return CollectionVersion object, latest or via query param 'version'."""
+        version = self.request.query_params.get('version', None)
```

Comment:
Would this be a reasonable place to set a 'lookup_field' on the viewset (ala, https://www.django-rest-framework.org/api-guide/generic-views/#attributes)

Reply:
hmm, nm. Given all the other cases to handle, 'lookup_field' isn't super useful